pkg/sink/sink.go

```diff
@@ -59,27 +62,48 @@ func (r Resource) HandleEvent(response http.ResponseWriter, request *http.Reques
 	el, err := r.TriggersClient.TektonV1alpha1().EventListeners(r.EventListenerNamespace).Get(r.EventListenerName, metav1.GetOptions{})
 	if err != nil {
 		log.Printf("Error getting EventListener %s in Namespace %s: %s", r.EventListenerName, r.EventListenerNamespace, err)
+		response.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
 	event, err := ioutil.ReadAll(request.Body)
 	if err != nil {
 		log.Printf("Error reading event body: %s", err)
+		response.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
 	eventId := template.Uid()
 	log.Printf("EventListener: %s in Namespace: %s handling event (EventID: %s) with payload: %s and header: %v",
 		r.EventListenerName, r.EventListenerNamespace, eventId, string(event), request.Header)
+
 	// Execute each Trigger
 	for _, trigger := range el.Spec.Triggers {
-		go r.executeTrigger(event, request.Header, trigger, eventId)
+		go r.executeTrigger(event, request, trigger, eventId)
 	}
-	fmt.Fprintf(response, "EventListener: %s in Namespace: %s handling event (EventID: %s)",
-		r.EventListenerName, r.EventListenerNamespace, eventId)
+
+	// TODO: Do we really need to return the entire body back???
+	fmt.Fprintf(response, "EventListener: %s in Namespace: %s handling event (EventID: %s) with payload: %s and header: %v",
+		r.EventListenerName, r.EventListenerNamespace, string(eventId), string(event), request.Header)
 }
 
-func (r Resource) executeTrigger(payload []byte, header http.Header, trigger triggersv1.EventListenerTrigger, eventId string) {
+func (r Resource) executeTrigger(payload []byte, request *http.Request, trigger triggersv1.EventListenerTrigger, eventId string) {
+	header := request.Header
+	if trigger.Interceptor != nil {
+		interceptorUrl, err := GetURI(trigger.Interceptor.ObjectRef, r.EventListenerNamespace) // TODO: Cache this result or do this on initialization
+		if err != nil {
+			log.Printf("Could not resolve Interceptor Service URI: %w", err)
+			return
+		}
+
+		modifiedPayload, err := r.processEvent(interceptorUrl, request, payload)
```

Comment:
I suppose an interceptor can intercept and augment the payload however they see fit. However, it seems to enforce that a payload should be returned?

Reply:
Right now, the contract is the service always has to return a body (adding docs). If the body should remain unmodified, simply return it back. We could potentially make this optional I guess?
1. If a body is returned use that as the event Payload
2. If not, just use the original body as the event Payload

Reply:
I think either approach makes sense. 👍